### PR TITLE
auth migration docs

### PR DESCRIPTION
**Pull request by beggers**

### CODE REVIEW

File: docs/migration.md [hunk 1]
```diff
@@ -22,6 +22,133 @@ We will aim to provide:
...
+
+- `chroma_server_auth_provider`
+- `chroma_server_auth_configuration_provider`
+- `chroma_server_auth_credentials_provider`
+- `chroma_client_auth_credentials_provider`
+- `chroma_client_auth_protocol_adapter`
+
+and others.
+
+We have consolidated all of this complexity into three classes:
```
Comment (HammadB) [inline docs/migration.md:44]:
```suggestion
We have consolidated these into three classes:
```

File: docs/migration.md [hunk 2]
```diff
@@ -22,6 +22,133 @@ We will aim to provide:
...
+
+- `ClientAuthProvider`
+- `ServerAuthenticationProvider`
+- `ServerAuthorizationProvider`
+
+`ClientAuthProvider`s are now responsible for their own configuration and credential management. Credentials can be given to them with the `chroma_client_auth_credentials` setting.
```
Comment (HammadB) [inline docs/migration.md:50]: dumb reader q - can chroma_client_auth_credentials be anything?

File: docs/migration.md [hunk 3]
```diff
@@ -22,6 +22,133 @@ We will aim to provide:
...
+`ServerAuthenticationProvider`s are responsible for turning a request's authorization information into a `UserIdentity` containing any information necessary to make an authorization decision. They are now responsible for their own configuration and credential management. Configured via the `chroma_server_authn_credentials` and `chroma_server_authn_credentials_file` settings.
```
Comment (HammadB) [inline docs/migration.md:51]: dumb_reader question what is credentials vs credentials_file?

File: docs/migration.md (hunk 2 above)
Comment (beggers) [reply docs/migration.md:50]: Added :^)

File: docs/migration.md (hunk 3 above)
Comment (beggers) [reply docs/migration.md:51]: Great question! I realized this should actually go in the Usage Guide which I missed before. Updated the usage guide and pointed to it from here.

**Pull request merged**